src/App.tsx

```diff
@@ -15,36 +19,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  name = 'Name',
+  length = 'Length',
+}
+
+interface SortProperty {
+  sortField: string;
+  isReversed: boolean;
+}
+
+function getPrepareGoods(
+  goods: Goods,
+  { sortField, isReversed }: SortProperty,
+): Goods {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      if (sortField.includes(SortType.name)) {
+        return good1.localeCompare(good2);
+      }
+
+      if (sortField.includes(SortType.length)) {
+        return good1.length - good2.length;
+      }
+
+      return 0;
+    });
+  }
+
+  if (isReversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
```

Comment:
```suggestion
  const [sortField, setSortField] = useState<SortType>('');
```
you can add an additional sortType to the enum, or make this state `SortType | null` with null by default